modules/core/src/main/scala/io/circe/jackson/snowplow/package.scala

```diff
@@ -45,7 +45,16 @@ package object snowplow {
           number match {
             case _: JsonBiggerDecimal | _: JsonBigDecimal =>
               number.toBigDecimal
-                .map(bigDecimal => DecimalNode.valueOf(bigDecimal.underlying))
+                .map(bigDecimal => {
+                  if (bigDecimal.isValidInt)
+                    IntNode.valueOf(bigDecimal.intValue)
+                  else if (bigDecimal.isValidLong) {
+                    LongNode.valueOf(bigDecimal.longValue)
+                  } else if (bigDecimal.isWhole) {
+                    BigIntegerNode.valueOf(bigDecimal.toBigInt.underlying)
+                  } else
+                    DecimalNode.valueOf(bigDecimal.underlying)
+                })
```

Comment:
Hi @hamnis please can you help me understand why this change is needed?  Is it related to the scala 3 upgrade, or is for some other reason?  Are there any examples of when the old code would give the wrong result?

I'm happy to merge it, along with your other changes, but I like to understand the reason for things first.

Reply:
It seems like circe on scala3 sometimes parses numbers differently, and I had to do this to avoid representing numbers in jackson incrorrectly. The tests where failing without this code.

I can try to find the actual reason in circe if you want.

if you want you can comment out this block and see how the tests fail.